File: src/main/java/org/dxworks/githubminer/dto/response/repository/branches/Branch.java
package org.dxworks.githubminer.dto.response.repository.branches;

import com.google.api.client.util.Key;
import lombok.Data;
import lombok.experimental.Delegate;

@Data
public class Branch {
    @Key
    private String name;
    @Key
    @Delegate
    private BranchCommit commit;
    @Key("protected")
    private boolean isProtected;
}

File: src/main/kotlin/org/dxworks/githubminer/dto/response/repository/branches/Branch.kt
package org.dxworks.githubminer.dto.response.repository.branches

import com.google.api.client.util.Key

class Branch {
    @Key
    var name: String? = null

    @Key
    var commit: BranchCommit? = null

    @Key("protected")
    var isProtected = false

    val sha
        get() = commit?.sha
}
